xbmc/music/infoscanner/MusicInfoScanner.cpp

```diff
@@ -112,7 +104,14 @@ void CMusicInfoScanner::Process()
         m_handle = dialog->GetHandle(g_localizeStrings.Get(314));
     }
 
-    m_bClean = g_advancedSettings.m_bMusicLibraryCleanOnUpdate;
+    // check if we only need to perform a cleaning
+    if (m_bClean && m_pathsToScan.empty())
+    {
+      CleanDatabase(false);
+      m_bRunning = false;
+
+      return;
```

Comment:
In general, early returns are bad. Here, there are side effects from the other code path if we return half way through the function. This branch should be taken at the beginning of the function.